packages/zoe/src/contractFacet/vatRoot.js

```diff
@@ -8,51 +8,84 @@
 // `yarn build-zcfBundle`.
 
 import { Far } from '@endo/marshal';
+import { E } from '@endo/far';
 
 import '../../exported.js';
 import '../internal-types.js';
 
 import { makeZCFZygote } from './zcfZygote.js';
 
+const { details: X } = assert;
+
 /**
  * @param {VatPowers & { testJigSetter: TestJigSetter }} powers
- * @returns {{ executeContract: ExecuteContract}}
+ * @param {{contractBundleCap: BundleCap, zoeService: ZoeService, invitationIssuer: Issuer, privateArgs: any}} vatParameters
+ * @param {import('@agoric/vat-data').Baggage} baggage
  */
-export function buildRootObject(powers) {
+// * @returns {{ executeContract: ExecuteContract}}
+export async function buildRootObject(powers, vatParameters, baggage) {
   // Currently, there is only one function, `executeContract` called
   // by the Zoe Service. However, when there is kernel support for
   // zygote vats (essentially freezing and then creating copies of
   // vats), `makeZCFZygote`, `zcfZygote.evaluateContract` and
   // `zcfZygote.startContract` should exposed separately.
   const { testJigSetter } = powers;
+  const { contractBundleCap } = vatParameters;
+  assert(
+    contractBundleCap !== undefined,
+    X`expected vatParameters.contractBundleCap ${vatParameters}`,
+  );
+  let { zoeService, invitationIssuer } = vatParameters;
+  const didStart = baggage.has('DidStart');
+  if (!didStart) {
```

Comment:
consider flipping the conditional blocks so it's `if (didStart)`. lighter cognitive load.

Reply:
Usually yes, but I found it easier to read this way, since it matches the chronology. The first time through it takes the first branch. Otherwise you have to read the "when we restart" before the "first time setup" code.

Would it be an improvement to reverse the sense of the flag?
```suggestion
  const firstTime = !baggage.has('DidStart');
  if (firstTime) {
```

Reply:
I am on the fence about flipping the sense of the flag.

Reply:
Status quo is fine. Resolving.